projects/components/src/select/select.component.test.ts

```diff
@@ -146,12 +150,56 @@ describe('Select Component', () => {
     const optionElements = spectator.queryAll('.select-option', { root: true });
     spectator.click(optionElements[2]);
 
-    expect(onChange).toHaveBeenCalledTimes(1);
     expect(onChange).toHaveBeenCalledWith(selectionOptions[2].value);
     expect(spectator.query('.trigger-content')).toHaveText(selectionOptions[2].selectedLabel!);
     flush();
   }));
 
+  test('should set trigger-prefix-icon correctly', fakeAsync(() => {
+    spectator = hostFactory(
+      `
+    <ht-select [icon]="icon">
+      <ht-select-option *ngFor="let option of options" [label]="option.label" [value]="option.value" [icon]="option.icon" [iconColor]="option.iconColor">
+      </ht-select-option>
+    </ht-select>`,
+      {
+        hostProps: {
+          options: selectionOptions,
+          icon: 'select-level-icon'
+        }
+      }
+    );
+    spectator.tick();
+
+    // No selection -> select component level icon and no color
+    expect(spectator.debugElement.query(By.css('.trigger-prefix-icon')).componentInstance.icon).toBe(
```

Comment:
is there a reason to use debug element querying rather than spectator querying ? is it because spectator doesn't give you access to the component on root queries?

Reply:
yep. wanted to access component to verify the contents of it.